src/Core/Form/IdentifiableObject/DataHandler/AbstractCategoryFormDataHandler.php

```diff
@@ -0,0 +1,183 @@
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Form\IdentifiableObject\DataHandler;
+
+use PrestaShop\PrestaShop\Core\CommandBus\CommandBusInterface;
+use PrestaShop\PrestaShop\Core\Domain\Category\Exception\MenuThumbnailsLimitException;
+use PrestaShop\PrestaShop\Core\Domain\Category\ValueObject\CategoryId;
+use PrestaShop\PrestaShop\Core\Domain\Category\ValueObject\MenuThumbnailId;
+use PrestaShop\PrestaShop\Core\Image\Uploader\ImageUploaderInterface;
+use Symfony\Component\HttpFoundation\File\UploadedFile;
+
+abstract class AbstractCategoryFormDataHandler implements FormDataHandlerInterface
+{
+    /**
+     * @var CommandBusInterface
+     */
+    protected $commandBus;
+
+    /**
+     * @var ImageUploaderInterface
+     */
+    protected $categoryCoverUploader;
+
+    /**
+     * @var ImageUploaderInterface
+     */
+    protected $categoryThumbnailUploader;
+
+    /**
+     * @var ImageUploaderInterface
+     */
+    protected $categoryMenuThumbnailUploader;
+
+    /**
+     * @param CommandBusInterface $commandBus
+     * @param ImageUploaderInterface $categoryCoverUploader
+     * @param ImageUploaderInterface $categoryThumbnailUploader
+     * @param ImageUploaderInterface $categoryMenuThumbnailUploader
+     */
+    public function __construct(
+        CommandBusInterface $commandBus,
+        ImageUploaderInterface $categoryCoverUploader,
+        ImageUploaderInterface $categoryThumbnailUploader,
+        ImageUploaderInterface $categoryMenuThumbnailUploader
+    ) {
+        $this->commandBus = $commandBus;
+        $this->categoryCoverUploader = $categoryCoverUploader;
+        $this->categoryThumbnailUploader = $categoryThumbnailUploader;
+        $this->categoryMenuThumbnailUploader = $categoryMenuThumbnailUploader;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function create(array $data)
+    {
+        if (!isset($data['menu_thumbnail_images']) || count($data['menu_thumbnail_images']) > count(MenuThumbnailId::ALLOWED_ID_VALUES)) {
+            throw new MenuThumbnailsLimitException('Maximum number of menu thumbnails exceeded for new category');
+        }
+        $command = $this->createAddCategoryCommand($data);
+
+        /** @var CategoryId $categoryId */
+        $categoryId = $this->commandBus->handle($command);
+
+        /**
+         * In some cases in form menu_thumbnail_images can be disabled so value won't get here.
+         */
+        $menuThumbnailImages = $data['menu_thumbnail_images'] ?? [];
+        $this->uploadImages(
+            $categoryId,
+            $data['cover_image'],
+            $data['thumbnail_image'],
+            $menuThumbnailImages
+        );
+
+        return $categoryId->getValue();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function update($categoryId, array $data)
```

Comment:
Add type for $categoryId

Reply:
This implements FormDataHandlerInterface interface, which doesn't have type so can't change it.